pyo3-build-config/src/lib.rs

```diff
@@ -115,11 +115,13 @@ const HOST_CONFIG: &str = include_str!(concat!(env!("OUT_DIR"), "/pyo3-build-con
 #[doc(hidden)]
 #[cfg(feature = "resolve-config")]
 fn resolve_cross_compile_config_path() -> Option<PathBuf> {
-    env::var_os("TARGET").map(|target| {
-        let mut path = PathBuf::from(env!("OUT_DIR"));
-        path.push(Path::new(&target));
-        path.push("pyo3-build-config.txt");
-        path
+    env::var_os("TARGET").and_then(|target| {
+        std::env::var_os("OUT_DIR").map(|out_dir| {
```

Comment:
Could you add a link to the documentation of e.g. bazel or buck2 that says how they set `OUT_DIR`? otherwise it will be hard to determine in the future why it's written in this way and how you may or may not change it

Reply:
I also think this strictly an improvement as we never needed compile-time access to `OUT_DIR` in the first place, i.e. we could have written it this way from the start. But let's wait what @konstin thinks w.r.t. your answer before merging.